src/category_theory/abelian/ext.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.category.Group.basic
+import algebra.category.Module.abelian
+import category_theory.derived
+import category_theory.linear.yoneda
+import category_theory.abelian.opposite
+import category_theory.abelian.projective
+
+/-!
+# Ext
+
+We define `ext R C n : Cᵒᵖ ⥤ C ⥤ Module R`
+for any `R`-linear abelian category `C`
+by left-deriving in the second argument of
+the `linear_yoneda` bifunctor `(X, Y) ↦ Module.of R (unop X ⟶ Y)`
+
+## Implementation
+It's not actually necessary here to assume `C` is abelian,
+but the hypotheses, involving both `C` and `Cᵒᵖ`, are quite lengthy,
+and in practice the abelian case is hopefully enough.
+-/
+
+noncomputable theory
+
+open category_theory
+
+variables (R : Type*) [ring R] (C : Type*) [category C] [abelian C] [linear R C]
+  [enough_projectives Cᵒᵖ] -- When we have injective resolutions, just `[enough_injectives C]`
+
+/--
+`ext R C n` is defined by left-deriving in the second argument of the `linear_yoneda` bifunctor.
+-/
+@[simps]
+def ext (n : ℕ) : Cᵒᵖ ⥤ C ⥤ Module R :=
```

Comment:
Should we capitalise this, to make very clear that it is not an extensionality gadget?

Reply:
Done.